from_json/date.ts

```
@@ -11,13 +11,78 @@ export function createDateStrategy(regex: RegExp): FromJsonStrategy {
   };
 }
 
-/** RegExp to test a string for a full ISO 8601 Date
- *  YYYY-MM-DDThh:mm:ss
- *  YYYY-MM-DDThh:mm:ssTZD
- *  YYYY-MM-DDThh:mm:ss.sTZD
- * @see: https://www.w3.org/TR/NOTE-datetime
- * @type {RegExp}
- */
-export const ISODateFromJson = createDateStrategy(
-  /^\d{4}-\d\d-\d\dT\d\d:\d\d:\d\d(\.\d+)?(([+-]\d\d:\d\d)|Z)?$/i,
-);
+/** converts timezone to minutes */
+function parseTimezone(zone: number): number {
+  return zone * 60 - Number.parseInt(`-${new Date().getTimezoneOffset()}`);
+}
+
+interface IsoWeekDate {
+  year: number;
+  week: number;
+  weekday: number;
+  timezone: number;
+}
+
+/** Test a string for a full ISO 8601 Date */
+export function iso8601Date(input: JsonValue): any {
+  var iso =
+    /^(\d{4})(?:-?W(\d+)(?:-?(\d+)D?)?|(?:-(\d+))?-(\d+))(?:[T ](\d+):(\d+)(?::(\d+)(?:\.(\d+))?)?)?(?:Z(-?\d*))?$/;
+  // throw error on bad data?
+  if (typeof input !== "string" || !iso.exec(input)) {
+    return input;
+  }
```

I think we should throw an error here instead of returning the input, or at least return `undefined`.

If we eventually add pre-parse validation we should have a proper way of validating this and handling errors, but for now, I think we should just try to avoid returning different types when errors occur.

Another option would be to simply return an `Invalid Date` type date, not sure how we would initialize that though.

---

Throwing an Error would be kind of how Jackson handles invalid input in Java.

---

done